### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by mogren**
*Issue #, if available:* #448

*Description of changes:*
* Add README for cni-metrics-helper
* Move images to the image folder
* Update troubleshoot guide

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cni-metrics-helper/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# cni-metrics-helper
+
+The `cni-metrics-helper` is a tool that can be used to scrape the aggregated ENIs and IPs information at the cluster
```
Comment (tiffanyfay) [inline cni-metrics-helper/README.md]: I think they should have '

File: cni-metrics-helper/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# cni-metrics-helper
+
+The `cni-metrics-helper` is a tool that can be used to scrape the aggregated ENIs and IPs information at the cluster 
+level and publish the metrics to CloudWatch. In able to publish metrics with useful names, the following two permissions
```
Comment (tiffanyfay) [inline cni-metrics-helper/README.md]: "The following two IAM permissions are required by the worker nodes to publish metrics and provide useful metric names:"

?

File: cni-metrics-helper/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+I0516 17:11:58.489648       7 metrics.go:350] Produce GAUGE metrics: ipamdActionInProgress, value: 0.000000
+I0516 17:11:58.489657       7 metrics.go:350] Produce GAUGE metrics: assignIPAddresses, value: 2.000000
+I0516 17:11:58.489665       7 metrics.go:350] Produce GAUGE metrics: totalIPAddresses, value: 11186.000000
+I0516 17:11:58.489674       7 metrics.go:350] Produce GAUGE metrics: eniMaxAvailable, value: 800.000000
+I0516 17:11:58.489685       7 metrics.go:340] Produce COUNTER metrics: ipamdErr, value: 1.000000
+I0516 17:11:58.489695       7 metrics.go:350] Produce GAUGE metrics: eniAllocated, value: 799.000000
+I0516 17:11:58.489715       7 metrics.go:350] Produce GAUGE metrics: maxIPAddresses, value: 11200.000000
+```
+
+## How do build
```
Comment (tiffanyfay) [inline cni-metrics-helper/README.md]: s/do/to

File: docs/troubleshooting.md [hunk 4]
```diff
@@ -2,23 +2,32 @@
 
 
 ## Manage Pod's IP address pool at cluster level
-As described in [Proposal: CNI plugin for Kubernetes networking over AWS VPC](./cni-proposal.md), ipamD allocates ENIs and secondary IP addresses from the instance subnet.
+As described in [Proposal: CNI plugin for Kubernetes networking over AWS VPC](./cni-proposal.md), ipamD allocates ENIs and 
+secondary IP addresses from the instance subnet.
 
-A node, based on the instance type ([Limit](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html#AvailableIpPerENI)), can have up to N ENIs and M addresses. The maximum number of IP addresses available to Pods on this node is min((N*M -N), subnet's free IP)
+A node, based on the instance type [limit](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html#AvailableIpPerENI), 
+can have up to N ENIs and M addresses. The maximum number of IP addresses available to Pods on this node is min((N*M -N), subnet's free IP)
```
Comment (tiffanyfay) [inline docs/troubleshooting.md]: > The maximum number of IP addresses available to Pods on this node is min((N*M -N))_, subnet's free IP)_

What? (the italics)

File: cni-metrics-helper/README.md (hunk 1 above)
Comment (tiffanyfay) [reply cni-metrics-helper/README.md]: Or maybe aggregated ENI and IP information. Can't grammar. Line 24 too in troubleshooting.md

File: docs/troubleshooting.md [hunk 5]
```diff
@@ -2,23 +2,32 @@
...
 
 ### Tip: Make sure there are enough ENIs and IPs for Pods in the cluster
 
-We provide a tool [**cni-metrics-helper**](../config/v1.4/cni-metrics-helper.yaml) which can show aggregated ENIs and IPs information at the cluster level.
+We provide a tool [**cni-metrics-helper**](../config/v1.4/cni-metrics-helper.yaml) which can show aggregated ENIs and IPs 
+information at the cluster level.
 
-You can optionally push them to cloudwatch.  For example:
+By default these metrics will be pushed to CloudWatch, but it can be disabled by setting `USE_CLOUDWATCH` to `"no"`. 
+This requires the `"cloudwatch:PutMetricData"` permission to publishe the metrics.
```
Comment (tiffanyfay) [inline docs/troubleshooting.md]: s/publishe/publish

Comment (mogren) [reply docs/troubleshooting.md]: Thanks, updated all.

File: cni-metrics-helper/README.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
+# cni-metrics-helper
+
+The `cni-metrics-helper` is a tool that can be used to scrape  ENI and IP information, aggregate it at the cluster
```
Comment (tiffanyfay) [inline cni-metrics-helper/README.md]: extra space after scrape XD

File: docs/troubleshooting.md [hunk 7]
```diff
@@ -2,23 +2,33 @@
...
 
 ## Manage Pod's IP address pool at cluster level
-As described in [Proposal: CNI plugin for Kubernetes networking over AWS VPC](./cni-proposal.md), ipamD allocates ENIs and secondary IP addresses from the instance subnet.
+As described in [Proposal: CNI plugin for Kubernetes networking over AWS VPC](./cni-proposal.md), ipamD allocates ENIs and 
+secondary IP addresses from the instance subnet.
 
-A node, based on the instance type ([Limit](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html#AvailableIpPerENI)), can have up to N ENIs and M addresses. The maximum number of IP addresses available to Pods on this node is min((N*M -N), subnet's free IP)
+A node, based on the instance type [limit](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html#AvailableIpPerENI), 
+can have up to N ENIs and M addresses. The maximum number of IP addresses available to Pods on this node is 
+`min((N * M - N), Free IP's in the subnet)`.
```
Comment (tiffanyfay) [inline docs/troubleshooting.md]: *free IPs

Comment (tiffanyfay) [reply docs/troubleshooting.md]: Also why is Pods capitalized?

**Pull request merged**